src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.CommonSVTControls.ps1

```diff
@@ -49,6 +58,32 @@ class CommonSVTControls: ADOSVTBase {
         return $controlResult
     }
 
+    hidden [ControlResult] CheckRepositoryPipelinePermission([ControlResult] $controlResult)
+    {
+        $controlResult.VerificationResult = [VerificationResult]::Failed
+        try
+        {
+            $projectId = ($this.ResourceContext.ResourceId -split "project/")[-1].Split('/')[0]
+            $url = "https://dev.azure.com/{0}/{1}/_apis/pipelines/pipelinePermissions/repository/{2}.{3}" -f $this.OrganizationContext.OrganizationName, $projectId, $projectId, $this.ResourceContext.ResourceDetails.Id;
+            $repoPipelinePermissionObj = @([WebRequestHelper]::InvokeGetWebRequest($url));
+
+            if (($repoPipelinePermissionObj.Count -gt 0) -and ([Helpers]::CheckMember($repoPipelinePermissionObj[0], "allPipelines")) -and ($repoPipelinePermissionObj[0].allPipelines.authorized -eq $true))
```

Comment:
If $repoPipelinePermissionObj.Count  = 0, is this a valid condition in some cases?
If not, then we should set result to error if count is 0

Reply:
Updated, removed first condition